documentation/sdk-manual/extensible.rst

```diff
@@ -677,6 +677,22 @@ For a recipe which inherits cmake it does:
 For a recipe which inherits meson a similar configuration is generated.
 Because there is nothing like a meson-preset a wrapper script for meson is generated.
 
+For remote debugging to work properly, the target must be running the corresponding image.
+You should also make sure that all dependencies for the recipe, as well as gdbserver
+are installed. Moreover, the target should be accessible through SSH. Lastly,
+you must make sure that the debug symbols are available in the sysroot. Here's
+an example bitbake fragment with the necessary configuration:
+
+.. code-block::
+
+   CORE_IMAGE_EXTRA_INSTALL += "gdbserver"
+
+   EXTRA_IMAGE_FEATURES += "ssh-server-openssh"
+   EXTRA_IMAGE_FEATURES += "debug-tweaks"
+
+   IMAGE_GEN_DEBUGFS = "1"
```

Comment:
# Remote debugging works much better if the binaries are in the rootfs as well.
IMAGE_CLASSES += "image-combined-dbg"

My setup looks like that:

```
# Build the companion debug file system
IMAGE_GEN_DEBUGFS = "1"
# But with devtool ide the dbg tar is not needed
IMAGE_FSTYPES_DEBUGFS = ""

# ssh is mandatory, no password simplifies the usage
EXTRA_IMAGE_FEATURES += "\
    ssh-server-openssh \
    debug-tweaks \
"

# Remote debugging needs the gdbserver on the target device
IMAGE_INSTALL:append = " gdbserver"
```

Reply:
Nice, I added the useful comments as well and removed the tar DEBUGFS.
Could you elaborate on what's not working well without image-combined-dbg? The debug symbols are not found when cross debugging?

Reply:
Yes if I remember correctly, gdb needs the binaries as well to find some debug symbols.

I have now revised the doc patch. I hope now everything is included and it is better understandable. Also the shared sysroot and cmake-kit part should now be covered.